samples/net/mqtt_publisher/src/main.c

```diff
@@ -36,12 +36,17 @@ static bool connected;
 
 #include "test_certs.h"
 
+#if (defined(CONFIG_NET_SOCKETS_OFFLOAD) && !defined(CONFIG_MBEDTLS))
+#define TLS_SNI_HOSTNAME "test.mosquitto.org"
```

Comment:
It indeed is connecting to a public server - the destination IP address changed as well.
The question that comes to my mind is, why to limit it only for offloading use-cases? If this is a valid test server for MQTT, we should be able to connect with native IP stack as well. Hence I think we should either make this sample configurable (to select between local/public broker) or provide separate sample.

Reply:
I admit I found the initial sample a bit confusing - It didn't have any TLS-specific instructions. There is a certificate in `test_certs.h`, with no indication of where it came from. It seems to expect prior knowledge from the user to know how to setup tls on their local mqtt server. I feel out-of-box the example should just work with as little effort from the user as possible, as opposed to imposing work on the user to figure out how to set up an mqtt server, which is partly why I just setup the hostname.

Another reason is that cc3220sf_launchxl expects the certificate in a file. It is much easier to just download a file then to create one from scratch, again lowering the bar to run the example for the first time.

That said I'd be happy to back out this hostname/address change and submit a different PR for 'ease of use', and for now leave it vague for the user to figure out how to set up a local server and create a certificate.

Reply:
> It seems to expect prior knowledge from the user to know how to setup tls on their local mqtt server. 

I'm with you here. In all fairness, I didn't test MQTT samples myself when reviewing it - because there are no instructions on how to setup reference testing environment (e.g. which mosuitto version to use, with which config, etc.). I have my mosquitto setup, but if doesn't work for example - it becomes problem of details of my setup, and digging with it not going to help other Zephyr users, so I skip it altogether, making a leap of faith that the original developer tested it all well. That's not good, and we need to resolve that (eventually), having a very specific test plans.

Here, my concern is different: we should not have samples behaving differently for different options of "non-constraining" variety. So, being able to run sample against test.mosquitto.org is absolutely great, but please make it available for all options, not just offloading. The opposite is also true - I understand there's extra legwork in setting up cc3220 for local server, but certainly TI's customers will want to do that, so please keep that on your radar, the sooner it's done, the higher chance it will be tested and elaborated before shipped to "production" customers (I of course don't call to do it right now).

All in all, I think @rlubos' suggestion is the best one - please make "test.mosquitto.org" vs "localhost" setup depend on its own config setting, and I'm +1 for having "test.mosquitto.org" the default.

Reply:
I have rolled back the changes to make all devices connect to a local server in order to be consistent. Let's address the connection to test.mosquitto.org in a different PR. I am thinking it can simply be a device-independent doc change, by adding a section in the README with the steps to connect to the public server.